app/scripts/components/common/blocks/chart/utils.js

```diff
@@ -1,12 +1,13 @@
-import { interpolateViridis } from 'd3-scale-chromatic';
+import * as d3ScaleChromatic from 'd3-scale-chromatic';
 
 export const fileExtensionRegex = /(?:\.([^.]+))?$/;
 
 export const chartMargin = { top: 50, right: 10, bottom: 100, left: 60 };
 export const itemHeight = 20;
 export const itemWidth = 120;
 
-function getLegendData({ data, width, itemWidth }) {
+export function getLegendConfig({ data, isMediumUp, colors, colorScheme }) {
+  const width = isMediumUp ? 600 : 450;
```

Comment:
Would it be possible for this to not be a hardcoded value? Maybe getting it dynamically?
Wondering because the min screen size can be 320px.

Reply:
Yeah, hard coded value is not ideal but it is not very straightforward to make style dynamic with svg texts. To get the dynamic width, (as far as I can think of) I would need to calculate the legend text width with paddings which I feel like bit more than I want to put a width for an element, so I thought it is better just to offer a width for texts rather than fitting width to texts.

Reply:
Ok. We should account for the min width possible as well then